jpf-symbc/build.gradle

```diff
@@ -0,0 +1,192 @@
+    // Gradle is able to infer the order of the source sets
+    // due to the compileClasspath attribute
+    dependsOn compileExamplesJava
+    dependsOn compileTestJava
+}
+
+task createJpfClassesJar(type: Jar) {
+    archiveName = "jpf-symbc-classes.jar"
+    destinationDir = file("${buildDir}")
+
+    group = "SPF Jars"
+    description = "Creates the ${archiveName} file."
+
+    dependsOn compile
+    dependsOn copyResources
+
+    from sourceSets.main.java.outputDir
+    from sourceSets.annotations.java.outputDir
+    from sourceSets.classes.java.outputDir
+}
+
+task createJpfJar(type: Jar) {
+    archiveName = "jpf-symbc.jar"
+    destinationDir = file("${buildDir}")
+
+    group = "SPF Jars"
+    description = "Creates the ${archiveName} file."
+
+    dependsOn compile
+    dependsOn copyResources
+
+    from sourceSets.main.java.outputDir
+    from sourceSets.peers.java.outputDir
+    from sourceSets.annotations.java.outputDir
+    from sourceSets.classes.java.outputDir
+
+    manifest {
+        attributes "Implementation-Title": "Symbolic Pathfinder system"
+        from manifestCommonAttrbutes
+    }
+}
+
+task createAnnotationsJar(type: Jar) {
+    archiveName = "jpf-symbc-annotations.jar"
+    destinationDir = file("${buildDir}")
+
+    group = "SPF Jars"
+    description = "Creates the ${archiveName} file."
+
+    dependsOn compile
+    dependsOn copyResources
+
+    from sourceSets.annotations.java.outputDir
+}
+
+task buildJars {
+    group = "SPF Build"
+    description = "Generates all JPF symbc jar files."
+
+    dependsOn createAnnotationsJar
+    dependsOn createJpfClassesJar
+    dependsOn createJpfJar
+}
+
+test {
+    description = "Runs core regression tests."
+
+    dependsOn buildJars
+
+    forkEvery = 1
+    enableAssertions = true
+    maxHeapSize = "1024m"
+
+    exclude "**/JPF_*.class"
+    include "**/Test*.class"
+    exclude "**/TestBitwise*.class"
+    exclude "**/TestCoverage.class"
+    exclude "**/TestDIV.class"
+    exclude "**/TestExJPF.class"
+    exclude "**/TestLazy*.class"    
+    exclude "**/TestPathCondition.class"
+    exclude "**/strings/*.class"
+    exclude "/gov/nasa/jpf/symbc/strings/*.class"
+    exclude "**/Test*\$*.class"
+    exclude "**/TestSymbolicListener.class"
+    include "**/TestSymbolicOutput.class"
+    include "**/TestSymbolicJPF.class"
+
+    testLogging {
+        events "passed", "skipped", "failed"
+    }
+
+    afterSuite { testDescriptor, result ->
+        if (!testDescriptor.parent) {
+            println "Test Execution: ${result.resultType}"
+
+            def summaryFields = ["${result.testCount} tests",
+                                 "${result.successfulTestCount} passed",
+                                 "${result.failedTestCount} failed",
+                                 "${result.skippedTestCount} skipped"]
+
+            println "Summary: " + summaryFields.join(", ")
+        }
+    }
+}
+
+publishing {
```

Comment:
Add a comment about where jars will be published. Generally, the file could be documented more.

Reply:
Done